### PR TITLE
Update Jet Runner to Jet version 4.0

### PR DESCRIPTION
**Pull request by jbartok**
Hazelcast Jet recently released version 4.0 (https://jet-start.sh/blog/2020/03/02/jet-40-is-released) and this pull request updates Beam's Jet Runner to that version. It also extends the Runners web page with information about the 4.0 version.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by jbartok** [comment]
R: @mxm

File: website/src/documentation/runners/jet.md [hunk 1]
```diff
@@ -59,24 +59,58 @@ Issue following command in the Beam examples project to start new Jet cluster an
...
 
 ### Running WordCount on a Remote Jet Cluster ##
-Download latest stable Hazelcast Jet code from [Hazelcast Website](https://jet.hazelcast.org/download/) and 
-start Jet cluster. 
-The simplest way is to start Jet cluster member using the `jet-start` script that comes with Jet distribution.
-The members use the [auto discovery feature](https://docs.hazelcast.org/docs/3.12/manual/html-single/index.html#setting-up-clusters) 
+Download latest stable Hazelcast Jet version, that Beam depends on, from 
+[Hazelcast Jet Website](https://jet-start.sh/download). At the time of writing the 
+[latest Beam release]({{site.baseurl}}/get-started/downloads/#releases) is 2.19.0, which still depends on Hazelcast
+Jet 3.0, but following versions should migrate to Jet 4.0.
```
Comment (mxm) [inline website/src/documentation/runners/jet.md]: Should we include a version table here similar to the Flink Runner page? I would not hardcode anything regarding the Beam release at the time of the writing.

Comment (jbartok) [reply website/src/documentation/runners/jet.md]: Did so, much better indeed. Thanks

**Comment by mxm** [comment]
Btw, there doesn't appear to be a Jenkins job (see .testinfra) for running the validates runner or other integration tests. Do you plan to add one?

**Comment by jbartok** [comment]
> Btw, there doesn't appear to be a Jenkins job (see .test-infra) for running the validates runner or other integration tests. Do you plan to add one?

I don't plan to add one, because I'm completely unfamiliar with the whole topic. I do have some Gradle targets in the Jet Runner module to execute @ValidatesRunner & @NeedsRunner tests with it, but that's it. I did check those, just as many pass now as before the upgrade.

**Comment by mxm** [comment]
I think it would be helpful to compare the jet runner with other runners. If there is already a Gradle target, it is very easy to do. The Jenkins job dsl scripts in `.test-infra/jenkins` automatically create the Jenkins job. You can see how they work if you just look at the ones for Flink or other runners.

**Comment by mxm** [comment]
Btw, the JIRA is missing, I've created it here: https://jira.apache.org/jira/browse/BEAM-9654

**Comment by jbartok** [comment]
Question: the runner's page (https://beam.apache.org/documentation/runners/jet/) has an "Adapt for" button, to change content between Jet 3.0 and Jet 4.0. It worked perfectly in a locally served website, but looks like it doesn't in the deployed one.... Any idea why?

**Comment by jbartok** [comment]
Yeah, works in the merged in version too, locally, but what ended up deployed is broken... No idea why.

**Comment by mxm** [comment]
Works for me.

**Comment by jbartok** [comment]
Yeah, now that I've flushed my browser cash works indeed... Thanks for checking.

**Pull request merged**